### PR TITLE
Added tests and docs to custom file class

### PR DESCRIPTION
**Pull request by DrPaulSharp**
Adds the class "testCustomFileClass" which sets out a suite of explicit tests for the custom file class within project class.
Adds custom RAT exceptions to utilities/exceptions and implements them in "customFileClass.m"
Tests for custom exceptions included in "testExceptions"

### CODE REVIEW

File: utilities/exceptions/invalidInput.m [hunk 1]
```diff
@@ -0,0 +1,10 @@
+classdef invalidInput < MException
```
Comment (StephenNneji) [inline utilities/exceptions/invalidInput.m:1]: What the difference btw invalid type and invalid input?

File: tests/testMultiTypeTable.m [hunk 2]
```diff
@@ -48,13 +48,13 @@
...
 
     properties
-        exampleTable;           % Example Multi-Type Table for testing
-        initialTypesTable;
-        initialAllowedTypes = {'constant'  'data'  'function', 'gaussian'};
-        initialAllowedActions = {'add'  'subtract'};
-        initialTypesCount = 1;
-        initialTypesAutoNameCounter = 1;
-        initialTypesAutoNameString = 'Row';
+        exampleTable            % Example Multi-Type Table for testing
```
Comment (StephenNneji) [inline tests/testMultiTypeTable.m:51]: I don't think there is any benefit to semicolon so we  have to agree a style an unify all

File: utilities/exceptions/invalidInput.m (hunk 1 above)
Comment (DrPaulSharp) [reply utilities/exceptions/invalidInput.m:1]: As I was developing the exceptions Invalid input split into invalid type and invalid number of inputs. I'll remove this one and we can always readd it later if needs be.

File: tests/testMultiTypeTable.m (hunk 2 above)
Comment (DrPaulSharp) [reply tests/testMultiTypeTable.m:51]: I'd spotted this as well. I'll take the semicolons out, which I prefer.

**Review by StephenNneji — APPROVED** [review]

**Pull request merged**